### PR TITLE
PSR1 & PSR2 support

### PR DESCRIPTION
**Pull request by shakaran**
This PR adds support for PSR1 and PSR2 spec using a .php_cs config file inside the repo (useful for run as git hook too).

Also update the code accordinly to the .php_cs specification to improve the code.

### CODE REVIEW

File: .php_cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
+<?php
+return PhpCsFixer\Config::create()
+    ->setRiskyAllowed(true)
+    ->setRules([
+    	'encoding' => true,
+        'full_opening_tag' => true,
```
Comment (jtojnar) [inline .php_cs]: Why not just use `'@PSR2' => true`  and overwrite only the modified options?

Comment (shakaran) [inline .php_cs]: Some PSR2 are not agreeded for everybody, making explicit each option help to easy switch or change it according the devs thoughs.

Comment (jtojnar) [inline .php_cs]: In either case, the file should be fixed by the fixer too, the indentation is inconsistent.

Comment (shakaran) [inline .php_cs]: Paradojic, but you are right. Commited the new changes

Comment (jtojnar) [inline .php_cs]: This way, however, it is not clear in which way is the coding style is modified.

**Comment by jtojnar** [comment]
Also you might want to add the checks to .travis.yml

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,35 @@
...
+  - composer install --prefer-source --no-interaction
+
 script:
-  - vendor/bin/phpunit
+  - mkdir -p build/logs
+  - vendor/bin/phpunit -c phpunit.xml.dist --verbose
+  - vendor/bin/php-cs-fixer fix --config=.php_cs --verbose --diff --dry-run
+
+after_script:
+  - wget https://scrutinizer-ci.com/ocular.phar
```
Comment (jtojnar) [inline .travis.yml]: Probably should be in the install section.

File: .travis.yml [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
+  - composer validate
+
+install:
+  - composer install --no-interaction --prefer-dist --no-progress --profile
+
 before_script:
-    - composer self-update
-    - composer install --prefer-source --no-interaction
+  - mkdir -p "$HOME/.php-cs-fixer"
+  - composer self-update
```
Comment (jtojnar) [inline .travis.yml]: This is also in before install.

File: .travis.yml [hunk 4]
```diff
@@ -1,12 +1,35 @@
...
+
 script:
-  - vendor/bin/phpunit
+  - mkdir -p build/logs
+  - vendor/bin/phpunit -c phpunit.xml.dist --verbose
+  - vendor/bin/php-cs-fixer fix --config=.php_cs --verbose --diff --dry-run
+
+after_script:
+  - wget https://scrutinizer-ci.com/ocular.phar
+  - php ocular.phar code-coverage:upload --format=php-clover build/logs/coverage.clover
```
Comment (jtojnar) [inline .travis.yml]: Does not seem to work.

**Comment by shakaran** [comment]
@jtojnar the new commit should install now the require dev (I will remove the binary download) and phpunit and php-cs-fixer should be available

**Comment by shakaran** [comment]
@jtojnar all should be ok now ;) (the test failing are because auth issues). Sorry for the noise

**Comment by abraham** [comment]
There is too much going on in this PR. I would likely accept most of it but it needs to be split into a number of discrete PRs.

**Pull request closed without merging**